doc/architecture/003-Implement-with-Rust.md

```diff
@@ -0,0 +1,108 @@
+# Implement with Rust
+
+## Context
+
+Due to significant architectural changes, NixOps 4 is a complete rewrite.
+This gives the project an opportunity to choose a different programming language.
+
+## Decision
+
+NixOps 4 will be implemented in Rust.
+
+- (+) General purpose language suitable for the purpose ("systems programming")
+- (+) Memory safety
+- (+) No garbage collector, makes integrating with a garbage-collected language easier
+- (+) Good performance
+- (+) Good Nix packaging, with multiple integrations available, which align with local development method
+- (+) Nix bindings available, although they will need to be improved, but they are mostly de-risked
```

Comment:
They existed but we're not using them...

Reply:
Hmm, I might not have been consistent about this, removing the AI line. Anyway, decision has been made so let's move on.